cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1927,21 +1928,46 @@ func (p *BicepProvider) ensureParameters(
 			continue
 		}
 
-		// For object inputs, if the "AZD Type" is a "inputs" it gets the value from the inputs section of the config,
-		// or an empty object, if there are no inputs configured.
+		// For object inputs, if the "AZD Type" is a "inputs" see if the autoGen inputs are already available in
+		// env config.
 		if p.mapBicepTypeToInterfaceType(param.Type) == ParameterTypeObject {
 			if m, has := param.AzdMetadata(); has && m.Type != nil && *m.Type == "inputs" {
-				var inputs map[string]any
-				if has, err := p.env.Config.GetSection("inputs", &inputs); err != nil {
+				existingInputs := make(map[string]map[string]any)
+				if _, err := p.env.Config.GetSection("inputs", &existingInputs); err != nil {
 					return nil, fmt.Errorf("reading inputs from config: %w", err)
-				} else if !has {
-					inputs = make(map[string]any)
 				}
+				wroteNewInput := false
 
-				configuredParameters[key] = azure.ArmParameterValue{
-					Value: inputs,
+				for inputResource, inputResourceInfo := range m.AutoGenerate {
+					existingRecordsForResource := make(map[string]any)
+					if current, exists := existingInputs[inputResource]; exists {
+						existingRecordsForResource = current
+					}
+					for inputName, inputInfo := range inputResourceInfo {
+						if _, has := existingRecordsForResource[inputName]; !has {
+							val, err := password.FromAlphabet(password.LettersAndDigits, inputInfo.Len)
+							if err != nil {
+								return nil, fmt.Errorf("generating value for input %s: %w", inputName, err)
+							}
+							existingRecordsForResource[inputName] = val
+							wroteNewInput = true
+						}
+					}
+					existingInputs[inputResource] = existingRecordsForResource
 				}
 
+				if wroteNewInput {
+					if err := p.env.Config.Set("inputs", existingInputs); err != nil {
+						return nil, fmt.Errorf("saving env config: %w", err)
+					}
+					if err := p.envManager.Save(ctx, p.env); err != nil {
+						return nil, fmt.Errorf("saving environment: %w", err)
+					}
+				}
+
+				configuredParameters[key] = azure.ArmParameterValue{
+					Value: existingInputs,
+				}
 				continue
 			}
 		}
```

Comment:
Looks like there is a good amount of complexity here that could benefit from some unit tests.  I'd also consider moving this out of the bicep provider and into its own bicep generation struct or similar.

Reply:
I'd recommend some refactoring here.  At this point this function now has 3 levels of nest for loops with multiple nested if blocks and this is not easy to follow at all.

Reply:
agree.
I brought the generate-impl from [service_target_dotnet_containerapp.go](https://github.com/Azure/azure-dev/pull/3448/files#diff-090fd77f1b0ee3651f9be718d203aa3b46b9b8593517343296d0ef83d588c6b8)

Let's try to make it simpler :)